delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViewsDeletion.scala

```diff
@@ -0,0 +1,63 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph
+
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.BlazegraphViews.{BlazegraphViewsAggregate, BlazegraphViewsCache}
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model.BlazegraphViewEvent
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model.BlazegraphViewRejection.ViewIsDeprecated
+import ch.epfl.bluebrain.nexus.delta.sdk.ResourcesDeletion.{CurrentEvents, ProjectScopedResourcesDeletion, StopActor}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourcesDeletionProgress.{CachesDeleted, ResourcesDataDeleted}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.ServiceAccount
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.DatabaseCleanup
+import monix.bio.Task
+
+final class BlazegraphViewsDeletion(
+    views: BlazegraphViews,
+    cache: BlazegraphViewsCache,
+    stopActor: StopActor,
+    currentEvents: CurrentEvents[BlazegraphViewEvent],
+    dbCleanup: DatabaseCleanup,
+    serviceAccount: ServiceAccount
+) extends ProjectScopedResourcesDeletion(stopActor, currentEvents, dbCleanup, BlazegraphViews.moduleType)(_.id) {
+
+  implicit private val subject: Subject = serviceAccount.subject
+
+  override def freeResources(projectRef: ProjectRef): Task[ResourcesDataDeleted] =
+    cache
+      .values(projectRef)
+      .flatMap { viewsList =>
+        Task.traverse(viewsList) { view =>
+          views
+            .deprecateWithoutProjectChecks(view.id, projectRef, view.rev)
```

Comment:
Did your tests go well with views ?
Because the deprecate event will be taken into account by the coordinator stream in (event-consistency-delay + refresh-interval) seconds.
What if the event is created and then deleted before the coordinator stream can read it ?
Maybe it would be safer to directly call the stream clean up from here ?

Reply:
The tests I did do delete the indices, so I assume it works. We could do further testing on this later on.